### ISSUE TITLE
LabelSampler does not yield balanced batches

### ISSUE THREAD

**Issue by fepegar** [fepegar/torchio#458]
**🐛Bug**

As reported by @malteekj in #457, label samplers are not working correctly. This happens because probability maps for voxel sampling are zeroed near the border and this is done after computing the voxel-wise sampling probability. Therefore, The probability of sampling the class near the border (background typically) will be smaller than it should be.

It is illustrated in [this gist](https://gist.github.com/fepegar/9e029c85827f48360ecc1b62b2530fa7).

**To reproduce**

```python
In [1]: import torch
   ...: import torchio as tio
   ...: 
   ...: size = 5
   ...: tensor = torch.zeros(1, size, size, 1)
   ...: tensor[0, 2, 2] = 1
   ...: label = tio.LabelMap(tensor=tensor)
   ...: 
   ...: subject = tio.Subject(label=label)
   ...: patch_size = 2, 2, 1
   ...: sampler = tio.LabelSampler(patch_size, label_probabilities={0: 1, 1:1})
   ...: values = torch.as_tensor([patch.label.data[0, 1, 1, 0] for patch in sampler(subject, 10000)])
   ...: print(values.mean())
tensor(0.6175)
```

whereas that value should have been much closer to 0.5.

**Expected behavior**

Getting an equal number of patch centers from background and foreground.

**Actual behavior**

The foreground is typically being oversampled. Not a terrible problem in practice, but this is not the correct behavior.

**TorchIO version**

`0.18.16`

**Comment by fepegar** [comment]
@all-contributors please add @malteekj for bug
